lib/collection.js

```diff
@@ -1874,6 +1891,57 @@ var count = function(self, query, options, callback) {
   });
 };
 
+/**
+ * Gets the number of documents matching the filter.
+ * @param {object} [query] the query for the count
+ * @param {object} [options=null] Optional settings.
+ * @param {object} [options.collation=null] Specifies a collation.
+ * @param {string|object} [options.hint=null] The index to use.
+ * @param {number} [options.limit=null] The maximum number of document to count.
+ * @param {number} [options.maxTimeMS=null] The maximum amount of time to allow the operation to run.
+ * @param {number} [options.skip=null] The number of documents to skip before counting.
+ * @param {Collection~countCallback} [callback] The command result callback.
+ * @return {Promise} returns Promise if no callback passed.
+ */
+
+Collection.prototype.countDocuments = function(query, options, callback) {
+  var args = Array.prototype.slice.call(arguments, 0);
+  callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;
+  query = args.length ? args.shift() || {} : {};
+  options = args.length ? args.shift() || {} : {};
+
+  return executeOperation(this.s.topology, countDocuments, [this, query, options, callback]);
+};
+
+const countDocuments = function(self, query, options, callback) {
+  const skip = options.skip;
+  const limit = options.limit;
+
+  const pipeline = [{ $match: query }];
+
+  // Add skip and limit if defined
+  if (typeof skip === 'number') {
+    pipeline.push({ $skip: skip });
+  }
+
+  if (typeof limit === 'number') {
+    pipeline.push({ $limit: limit });
+  }
+
+  pipeline.push({ $group: { _id: null, n: { $sum: 1 } } });
+
+  delete options.limit;
+  delete options.skip;
+
+  self.aggregate(pipeline, options, function(err, result) {
```

Comment:
Let's add a TODO here to look out for how this plays into the operations refactor. CC @kvwalker